### ISSUE TITLE
Property types and no-mergeable-namespace

### ISSUE THREAD

**Issue by schmuli** [palantir/tslint#2051]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: vscode-tslint

#### TypeScript code being linted

```ts
// app.ts

// Interface declared outside of any namespace
interface IExternal {
    prop: app.IProp;
}

namespace app {
    export interface IProp {
        ...
    }
}
```

with `tslint.json` configuration:

```json
{ "no-mergeable-namespace": true }
```

#### Actual behavior

The `no-mergeable-namespace` rule highlights the type of `IExternal::prop` as a mergeable namespace.

#### Expected behavior

The `no-mergeable-namespace` should ignore property types, as these cannot really be merged. The only other option is to define the `IExternal` interface in a separate file, which is overkill.
